fileinfo.go

```diff
@@ -36,6 +38,8 @@ func SetFileBasicInfo(f *os.File, bi *FileBasicInfo) error {
 	if err := setFileInformationByHandle(syscall.Handle(f.Fd()), fileBasicInfo, (*byte)(unsafe.Pointer(bi)), uint32(unsafe.Sizeof(*bi))); err != nil {
 		return &os.PathError{Op: "SetFileInformationByHandle", Path: f.Name(), Err: err}
 	}
+	runtime.KeepAlive(f)
+	runtime.KeepAlive(bi)
```

Comment:
This one I'm unsure about. After `(*byte)(unsafe.Pointer(bi))` returns the pointer, is `bi` still considered live? How do the new runtime argument liveness rules apply to cgo rules? `bi` is no longer accessible as soon as `unsafe.Pointer` returns, so based on my understanding, a finalizer could run on `bi` before the call to c.

It doesn't matter though, as `bi` does not have a finalizer on it, so I can remove this regardless, but I don't see any documents regarding the new liveness rules when passing pointers to cgo.